windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -0,0 +1,535 @@
+            OnPropertyChanged(nameof(ToolbarSeparatorLeft));
+        }
+    }
+
+    public bool ToolbarSeparatorLeft
+    {
+        get
+        {
+            var left = ToolbarRemove | ToolbarTrash | ToolbarCleanup;
+            var right = ToolbarResume | ToolbarStop | ToolbarReload;
+            return left && right;
+        }
+    }
+
+    public bool ToolbarSeparatorRight
+    {
+        get
+        {
+            return ToolbarOpen | ToolbarShow;
+        }
+    }
+
+    public ReadOnlyObservableCollection<TransferViewModel> Transfers => transfers;
+
+    public TransfersViewModel(
+        TransferController controller,
+        TransfersStore store,
+        Locale locale,
+        BandwidthProvider bandwidthProvider,
+        ConnectionProvider connectionProvider,
+        Preferences preferences,
+        WpfIconProvider iconProvider)
+    {
+        this.controller = controller;
+        this.store = store;
+        this.preferences = preferences;
+        this.iconProvider = iconProvider;
+
+        var localTransfers = store.Transfers.Connect().ObserveOnDispatcher()
+            .Transform(m => new TransferViewModel(controller, m, locale)).DisposeMany()
+            .Bind(out transfers)
+            .AutoRefresh(m => m.IsSelected).Filter(m => m.IsSelected).Bind(out selectedTransfers)
+            .AsObservableCache();
+        localTransfers.Connect().WhenAnyPropertyChanged().Subscribe(OnSelectedTransferPropertyChanged);
+        localTransfers.CountChanged.Subscribe(OnSelectionCountChanged);
+
+        BandwidthViewModel unlimitedBandwidth = new(-1, true, locale.localize("Unlimited Bandwidth", "Preferences"));
+        Bandwidth = [
+            unlimitedBandwidth,
+            ..bandwidthProvider.Bandwidth.Select(m => new BandwidthViewModel(m, true)),
+        ];
+        bandwidthLookup = Bandwidth.ToDictionary(m => m.Bandwidth);
+        selectedBandwidth = unlimitedBandwidth;
+
+        Connections = [
+            .. connectionProvider.Connections.Select(m => new ConnectionViewModel(m, true)),
+        ];
+        connectionLookup = Connections.ToDictionary(m => m.Connections);
+        selectedConnectionLimit = FetchConnectionLimit();
+    }
+
+    partial void OnSelectedConnectionLimitChanged(ConnectionViewModel value)
+    {
+        preferences.setProperty("queue.connections.limit", value.Connections);
+        TransferQueueFactory.get().resize(value.Connections);
+    }
+
+    partial void OnSelectedTransferChanged(TransferViewModel newValue)
+    {
+        UpdateSelectedBandwidth();
+        UpdateSelectionPanel();
+    }
+
+    private ConnectionViewModel FetchConnectionLimit()
+    {
+        var rate = preferences.getInteger("queue.connections.limit");
+        if (!connectionLookup.TryGetValue(rate, out var selection))
+        {
+            connectionLookup[rate] = selection = new(rate, false, null);
+            Connections.Add(selection);
+        }
+
+        return selection;
+    }
+
+    [RelayCommand]
```

Comment:
Mentioned once: RelayCommandAttribute from CommunityToolkit.Mvvm is a source generator, wiring up a lazy RelayCommand, which invokes OnClean in Execute, and may call CanExecute-parameter for validation whether the command should run.